### ISSUE TITLE
Icons Not Showing on Mac

### ISSUE THREAD

**Issue by diminutivesloop** [DanBrooker/file-icons#416]
I’m currently syncing my .atom folder between a Mac and a Windows PC using Dropbox. The icons display as expected on the Windows PC, but on the Mac almost all the icons display as empty boxes. I’ve tried reinstalling the package and restarting Atom, but the issue persists. I’ve included some screenshots to illustrate the problem.

![atom-file-icons-defect-1](https://cloud.githubusercontent.com/assets/667716/17502931/db3b6304-5db9-11e6-8a55-187db6028a69.png)
![atom-file-icons-defect-2](https://cloud.githubusercontent.com/assets/667716/17502937/dfe15a9e-5db9-11e6-9fd2-78057ecaaded.png)

**Comment by Alhadis** [comment]
Are there any network/load errors when you open the developer tools?

I can tell at a glance the icon font isn't loading correctly.

**Comment by Alhadis** [comment]
Actually... *none* of the icon fonts are loading except Octicons. :| What exactly are you syncing? The *entire* `.atom` user directory?

If so, it might be because your `compile-cache` is being shared with foreign path separators (I'm guessing).

**Comment by nerdgore** [comment]
I am having the same issue. It started a week or so ago. Usually setting another theme or disabling and then enabling file-icons solves this for the current session.

**Comment by Alhadis** [comment]
What version of Mac are you running, guys?

I'm running Mavericks and haven't experienced any such issues...

**Comment by nerdgore** [comment]
![screen shot 2016-08-09 at 11 05 07](https://cloud.githubusercontent.com/assets/1582533/17510947/58d01302-5e21-11e6-8032-5a0c63a11fa7.png)

I am running El Capitan. But this issue did not appear with an update if that helps. It also seems that after a very long time (several minutes) the icons appear on their own.

**Comment by Alhadis** [comment]
The hell. Can you [PasteBin](http://pastebin.com/) me the output of `apm list`...?

**Comment by nerdgore** [comment]
```
Built-in Atom Packages (89)
├── about@1.5.2
├── archive-view@0.61.1
├── atom-dark-syntax@0.27.0
├── atom-dark-ui@0.51.0
├── atom-light-syntax@0.28.0
├── atom-light-ui@0.43.0
├── autocomplete-atom-api@0.10.0
├── autocomplete-css@0.11.1
├── autocomplete-html@0.7.2
├── autocomplete-plus@2.31.0
├── autocomplete-snippets@1.11.0
├── autoflow@0.27.0
├── autosave@0.23.1
├── background-tips@0.26.0
├── base16-tomorrow-dark-theme@1.1.0
├── base16-tomorrow-light-theme@1.1.1
├── bookmarks@0.41.0
├── bracket-matcher@0.82.1
├── command-palette@0.38.0
├── deprecation-cop@0.54.1
├── dev-live-reload@0.47.0
├── encoding-selector@0.22.0
├── exception-reporting@0.38.3
├── find-and-replace@0.198.0
├── fuzzy-finder@1.3.0
├── git-diff@1.1.0
├── go-to-line@0.31.0
├── grammar-selector@0.48.1
├── image-view@0.58.0
├── incompatible-packages@0.26.1
├── keybinding-resolver@0.35.0
├── language-c@0.52.1
├── language-clojure@0.21.0
├── language-coffee-script@0.47.0
├── language-csharp@0.12.1
├── language-css@0.36.2
├── language-gfm@0.86.0
├── language-git@0.13.0
├── language-go@0.42.0
├── language-html@0.44.1
├── language-hyperlink@0.16.0
├── language-java@0.22.0
├── language-javascript@0.119.0
├── language-json@0.18.0
├── language-less@0.29.3
├── language-make@0.22.2
├── language-mustache@0.13.0
├── language-objective-c@0.15.1
├── language-perl@0.35.0
├── language-php@0.37.0
├── language-property-list@0.8.0
├── language-python@0.45.0
├── language-ruby@0.68.5
├── language-ruby-on-rails@0.25.0
├── language-sass@0.52.0
├── language-shellscript@0.22.3
├── language-source@0.9.0
├── language-sql@0.21.1
├── language-text@0.7.1
├── language-todo@0.28.0
├── language-toml@0.18.0
├── language-xml@0.34.8
├── language-yaml@0.26.0
├── line-ending-selector@0.5.0
├── link@0.31.1
├── markdown-preview@0.158.0
├── metrics@0.53.1
├── notifications@0.64.1
├── one-dark-syntax@1.2.0
├── one-dark-ui@1.3.2
├── one-light-syntax@1.2.0
├── one-light-ui@1.3.2
├── open-on-github@1.2.0
├── package-generator@1.0.0
├── settings-view@0.238.0
├── snippets@1.0.2
├── solarized-dark-syntax@1.0.2
├── solarized-light-syntax@1.0.2
├── spell-check@0.67.1
├── status-bar@1.4.1
├── styleguide@0.46.0
├── symbols-view@0.113.0
├── tabs@0.98.0
├── timecop@0.33.1
├── tree-view@0.208.0
├── update-package-dependencies@0.10.0
├── welcome@0.34.0
├── whitespace@0.32.2
└── wrap-guide@0.38.1

Community Packages (84)
├── Sublime-Style-Column-Selection@1.7.2
├── achievements@0.9.0
├── activate-power-mode@0.7.3
├── advanced-open-file@0.16.3
├── astromac-ui@0.1.0
├── atom-html-preview@0.1.22
├── atom-import-sort@0.6.0
├── atom-ternjs@0.14.2
├── atom-wallaby@1.0.9
├── atom-wrap-in-tag@0.6.0
├── autocomplete-html-entities@0.1.0
├── autocomplete-modules@1.6.1
├── autocomplete-snippets@1.11.0
├── ava@0.5.2
├── base16-retro-computers-syntax@0.0.0
├── base16-syntax@1.7.0
├── block-travel@1.0.5
├── coffee-refactor@0.7.0
├── dark-flat-ui@1.2.4
├── docblockr@0.7.4
├── editorconfig@1.4.1
├── emmet@2.4.3
├── es6-javascript@1.0.0
├── eslint-fixer@1.1.0
├── extract-selection@0.2.1
├── file-icons@1.7.18
├── git-blame@0.4.11
├── git-history@3.2.0
├── git-log@0.4.1
├── git-time-machine@1.5.3
├── gitkraken@0.4.1
├── gruvbox-plus-syntax@1.2.5
├── highlight-line@0.11.1
├── imdone-atom@1.3.29
├── imdone-atom-github@0.1.11
├── inline-autocomplete@1.1.0
├── js-refactor@0.7.3
├── json-converter@0.2.4
├── kary-foundation-dark@4.0.0
├── kary-foundation-light@4.0.0
├── language-babel@2.32.2
├── language-javascript-jsx@0.3.7
├── language-markdown@0.16.1
├── language-soy@0.3.2
├── lines@0.13.1
├── linter@1.11.16
├── linter-doiuse@0.2.3
├── linter-eslint@7.2.4
├── linter-jscs@4.0.5
├── linter-jshint@2.1.0
├── linter-jsonlint@1.3.0
├── linter-less@2.6.0
├── linter-lesshint@2.0.2
├── markdown-preview-plus@2.4.0
├── markdown-sort-list@0.2.0
├── merge-conflicts@1.4.4
├── minimap@4.24.7
├── minimap-pigments@0.2.1
├── native-ui@0.18.3
├── neutron-syntax@0.7.0
├── neutron-ui@0.4.0
├── node-resolver@1.0.1
├── npm-install@4.0.3
├── npm-outdated@0.5.0
├── open-recent@5.0.0
├── pigments@0.31.2
├── pull-requests@0.0.23
├── refactor@0.9.2
├── rest-client@1.2.1
├── run-in-atom@1.0.5
├── select-rectangle@1.0.2
├── seti-syntax@1.0.1
├── seti-ui@1.3.2
├── tablr@1.3.2
├── terminal-plus@0.14.5
├── tidy-markdown@3.0.1
├── tidy-tabs@0.3.0
├── todo-show@1.7.0
├── toggle-quotes@1.0.0
├── tree-view-git-status@1.1.0
├── turbo-javascript@1.4.0
├── twilight-syntax@0.7.0
├── twilight-syntax-spacegray-ui@0.4.0
└── twilight-textmate-syntax@0.1.19

└── (empty)
```

**Comment by Alhadis** [comment]
... erm, I did say PasteBin, but massive lists are cool too, I guess.

I'll have a look into this, cheers.

**Comment by nerdgore** [comment]
oops, sorry my bad

**Comment by Alhadis** [comment]
@nerdgore When this happens, what CSS properties are being applied to the icons? Are there any from this package? If you kick the dev tools open, you should see something like this when you inspect an icon in the tree-view:

<img width="600" alt="Figure 1" src="https://cloud.githubusercontent.com/assets/2346707/17511547/b98485bc-5e66-11e6-89ef-f9707032110f.png" />

Also, which themes do you have enabled when this happens? (Just trying to get as much info together as possible here)

**Comment by diminutivesloop** [comment]
Just checked the network tab in Chrome and found this little nugget.
![atom-file-icons-defect-3](https://cloud.githubusercontent.com/assets/667716/17537726/e53b509a-5e6c-11e6-98d1-89d4e95518c2.png)

**Comment by diminutivesloop** [comment]
I’m currently running OS X 10.11.6, and I’m syncing the entire .atom folder via Dropbox. Also I haven’t noticed the problem go away even if I toggle themes or leave Atom open for an extended period.

**Comment by Alhadis** [comment]
What the... can you try this?

1. Open [styles/fonts.less](https://github.com/DanBrooker/file-icons/blob/master/styles/fonts.less) in the downloaded package's directory
2. Replace `format("woff")` with `format(woff2)`
3. Quit and restart Atom

Does that make a difference?

**Comment by nerdgore** [comment]
Okay, I tried brute force and found out what causes icons to disappear for me.

(De)activating "js-refactor" I can reproduce the issue.

**Comment by Alhadis** [comment]
Perfect, now we're getting somewhere.

Installing that package now, I'll see if I can reproduce this.

**Comment by nerdgore** [comment]
In my case having js-refactor enabled is cancelling all requests to *.woff files. So it seems to be an issue with loading rather than styling. And that issue seems to be with another package.

![screen shot 2016-08-10 at 17 42 22](https://cloud.githubusercontent.com/assets/1582533/17560306/e323c3ea-5f21-11e6-83b3-2cbb90e403bb.png)

**Comment by Alhadis** [comment]
Not just woff files, it's cancelling requests to a lot of stuff. :| It's also blocked `octicons.woff` (which is supplied by Atom as a core resource) and also the metric collection requests Atom makes when collecting analytics.

This seems to be a pretty nasty issue with the js-refactor package, or something. You'd best file an issue there.

**Comment by Alhadis** [comment]
@diminutivesloop Does the network tab give you any insights if you click or hover on the font's entry? Nothing about *why* the resource wasn't loaded?

If that tells you nothing, what about the **Resources** tab?
<img src="https://cloud.githubusercontent.com/assets/2346707/17560589/0028fc14-5f66-11e6-9d3c-7b489087269c.png" width="283" alt="Figure 1" />

**Comment by diminutivesloop** [comment]
Looks like the issue is fixed for me after updating to Atom 1.9.8.

**Comment by Alhadis** [comment]
Hrm, strange... :S

Well, if it pops up again, feel free to open another issue.
